test/ClientFailoverTest.js

```diff
@@ -0,0 +1,258 @@
+                        getFn();
+                    }
+                })
+                .catch((err) => {
+                    getErr = err;
+                });
+        };
+        process.nextTick(getFn);
+        await RC.terminateCluster(cluster1.id);
+        await clusterChangedDeferred.promise;
+
+        expect(getErr).to.be.undefined;
+        expect(client.getCluster().getMembers()).to.have.lengthOf(1);
+        expect(client.getCluster().getMembers()[0].uuid.toString()).to.be.equal(member2.uuid);
+    });
+
+    it('should reject with offline error when retrying read ops while switching in async mode', async function () {
+        markEnterprise(this);
+
+        const config = createClientConfig({ clusterName: 'dev1', connectTimeoutMs: 1000 });
+        config.connectionStrategy.reconnectMode = 'ASYNC';
+        client = await Client.newHazelcastFailoverClient({
+            // default tryCount is Number.MAX_SAFE_INTEGER
+            clientConfigs: [ config ]
+        });
+
+        const map = await client.getMap('test');
+        const getErrDeferred = deferredPromise();
+        let getErr;
+        const getFn = () => {
+            map.get('foo')
+                .then(() => {
+                    if (!clusterChanged) {
+                        getFn();
+                    }
+                })
+                .catch((err) => {
+                    getErr = err;
+                    getErrDeferred.resolve();
+                });
+        };
+        process.nextTick(getFn);
+        await RC.terminateCluster(cluster1.id);
+        await getErrDeferred.promise;
+
+        expect(getErr).to.be.instanceOf(ClientOfflineError);
+    });
+
+    it('should shutdown when switching to cluster with different partition count', async function () {
+        markEnterprise(this);
+
+        cluster3 = await RC.createClusterKeepClusterName(null,
+            createClusterConfig({ clusterName: 'dev3', partitionCount: 42 }));
+        await RC.startMember(cluster3.id);
+
+        const shutdownTriggeredDeferred = deferredPromise();
+        const lifecycleListener = (state) => {
+            if (state === 'SHUTDOWN') {
+                shutdownTriggeredDeferred.resolve();
+            }
+        };
+        client = await Client.newHazelcastFailoverClient({
+            tryCount: 1,
+            clientConfigs: [
+                createClientConfig({ clusterName: 'dev1', lifecycleListener, connectTimeoutMs: 1000 }),
+                createClientConfig({ clusterName: 'dev3', lifecycleListener, connectTimeoutMs: 1000 })
+            ]
+        });
+
+        expect(client.getCluster().getMembers()).to.have.lengthOf(1);
+        expect(client.getCluster().getMembers()[0].uuid.toString()).to.be.equal(member1.uuid);
+
+        await RC.terminateCluster(cluster1.id);
+
+        await shutdownTriggeredDeferred.promise;
+    });
+
+    it('should throw when starting with invalid failover config', async function () {
+        markEnterprise(this);
```

Comment:
There is no problem right now but, can we mark the describe block as enterprise and single open-source server test as community? My reasoning is that we may forget to mark some new test as enterprise in the future, and that may cause problems in the compatibility test suite

Reply:
Good suggestion. Applied it in 3170874e8bee67d3fb3a1764861bc86b8387e3b1